docs/gallery/domain/ecephys.py

```diff
@@ -5,152 +5,291 @@
 Extracellular Electrophysiology Data
 ====================================
 
+The following tutorial describes storage of extracellular electrophysiology data in NWB.
+The workflow demonstrated here involves four main steps:
+
+1. Create the electrodes table
+2. Add acquired raw voltage data
+3. Add LFP data
+4. Add spike data
+
+
+This tutorial assumes that transforming data between these states is done by users--PyNWB does not provide
+analysis functionality. It is recommended to cover :ref:`basics` before this tutorial.
+
+The following examples will reference variables that may not be defined within the block they are used in. For
+clarity, we define them here:
 '''
 
+# sphinx_gallery_thumbnail_path = 'figures/gallery_thumbnails_ecephys.png'
+from datetime import datetime
+from dateutil.tz import tzlocal
+
+import numpy as np
+from pynwb import NWBFile, NWBHDF5IO
+from pynwb.ecephys import ElectricalSeries, LFP
+
 #######################
 # Creating and Writing NWB files
 # ------------------------------
 #
-# When creating a NWB file, the first step is to create the :py:class:`~pynwb.file.NWBFile`. The first
-# argument is the name of the NWB file, and the second argument is a brief description of the dataset.
+# When creating a NWB file, the first step is to create the :py:class:`~pynwb.file.NWBFile`.
 
-# sphinx_gallery_thumbnail_path = 'figures/gallery_thumbnails_ecephys.png'
-
-import numpy as np
-from datetime import datetime
-from dateutil.tz import tzlocal
-from pynwb import NWBFile
-
-nwbfile = NWBFile('my first synthetic recording', 'EXAMPLE_ID', datetime.now(tzlocal()),
-                  experimenter='Dr. Bilbo Baggins',
-                  lab='Bag End Laboratory',
-                  institution='University of Middle Earth at the Shire',
-                  experiment_description='I went on an adventure with thirteen dwarves to reclaim vast treasures.',
-                  session_id='LONELYMTN')
 
+nwbfile = NWBFile(
+    'my first synthetic recording',
+    'EXAMPLE_ID',
+    datetime.now(tzlocal()),
```

Comment:
Because most PyNWB constructors have so many arguments, to improve code readability, I recommend as best practice that we use and showcase keyword arguments rather than positional arguments.
```suggestion
    session_description='my first synthetic recording',
    identifier='EXAMPLE_ID',
    session_start_time=datetime.now(tzlocal()),
```